src/debug_renderer.rs

```diff
@@ -35,6 +39,67 @@ pub struct DebugRenderer<R: gfx::Resources, F: Factory<R>> {
     factory: F,
 }
 
+type WorldPosition = [f32; 3];
+type ScreenPosition = [i32; 2];
+type Color = [f32; 4];
+type Size = f32;
+
+enum DrawCommand {
+    DrawLine(WorldPosition, WorldPosition, Color),
+    DrawMarker(WorldPosition, Size, Color),
+    DrawScreenText(String, ScreenPosition, Color),
+    DrawWorldText(String, WorldPosition, Color),
+}
+
+#[derive(Clone)]
+struct CommandQueue {
+    queue: Arc<Mutex<VecDeque<DrawCommand>>>
+}
+
+impl CommandQueue {
+
+    fn push_back(&mut self, command: DrawCommand) {
+        let mut queue = self.queue.lock().unwrap();
+        queue.push_back(command);
+    }
+
+    fn pop_front(&mut self) -> Option<DrawCommand> {
+        let mut queue = self.queue.lock().unwrap();
+        queue.pop_front()
+    }
+
+    fn instance() -> CommandQueue {
+        static mut command_queue_singleton: *const CommandQueue = 0 as *const CommandQueue;
+        static ONCE: Once = ONCE_INIT;
+
+        unsafe {
+            ONCE.call_once(|| {
+                let command_queue = CommandQueue {
+                    queue: Arc::new(Mutex::new(VecDeque::<DrawCommand>::new()))
+                };
+                command_queue_singleton = mem::transmute(Box::new(command_queue));
```

Comment:
I think this is a memory leak, see http://stackoverflow.com/questions/27791532/how-do-i-create-a-global-mutable-singleton. Looking into it...